src/main/scala/com/databricks/spark/csv/package.scala

```diff
@@ -43,21 +43,23 @@ package object csv {
   }
 
   implicit class CsvSchemaRDD(dataFrame: DataFrame) {
-    def saveAsCsvFile(path: String): Unit = {
+    def saveAsCsvFile(path: String, parameters: Map[String, String] = Map()): Unit = {
       // TODO(hossein): For nested types, we may want to perform special work
-      val header = dataFrame.columns.map(c => s""""$c"""").mkString(",")
+      val delimiter = parameters.getOrElse("delimiter", ",")
+      val generateHeader = parameters.getOrElse("generateHeader", "false").toBoolean
```

Comment:
Note that in other cases `header` is referring to wether first row is header. Here it means whether we should generate a header for the file we output. Naming this `header` again might be confusing. Though that's up to you to decide. If you still thing `header` is good, I'll make the change.